iox_catalog/src/interface.rs

```diff
@@ -480,6 +480,16 @@ pub trait PartitionRepo: Send + Sync {
 
     /// List the records of compacting a partition being skipped. This is mostly useful for testing.
     async fn list_skipped_compactions(&mut self) -> Result<Vec<SkippedCompaction>>;
+
+    /// Update the per-partition persistence watermark.
+    ///
+    /// The given `sequence_number` is the inclusive maximum [`SequenceNumber`]
+    /// of the most recently persisted data for this partition.
+    async fn update_persist_watermark(
```

Comment:
```suggestion
    async fn update_persisted_sequence_number(
```

Is there a reason to use a different term (`persist_watermark`) in the catalog interface compared to the `Partition` structure?

Reply:
Because I renamed everything half way through :D

I prefer your suggestion, 2 secs :+1:

Reply:
1b57a4704